distributed/tests/test_scheduler.py

```diff
@@ -1068,6 +1068,30 @@ def test_service_hosts_match_scheduler():
         yield s.close()
 
 
+@pytest.mark.skipif(not sys.platform.startswith('linux'),
+                    reason="Need 127.0.0.* to mean localhost")
+@gen_test(timeout=None)
+def test_service_custom_host():
+    pytest.importorskip('bokeh')
+    from distributed.bokeh.scheduler import BokehScheduler
+    services = {('bokeh', ('127.0.0.3', 0)): BokehScheduler}
+
+    s = Scheduler(services=services)
+    yield s.start('tcp://0.0.0.0')
+
+    sock = first(s.services['bokeh'].server._http._sockets.values())
+    assert sock.getsockname()[0] == '127.0.0.3'
+    yield s.close()
+
+    for host in ['tcp://127.0.0.2', 'tcp://127.0.0.2:38275']:
+        s = Scheduler(services=services)
+        yield s.start(host)
+
+        sock = first(s.services['bokeh'].server._http._sockets.values())
+        assert sock.getsockname()[0] == '127.0.0.3'
+        yield s.close()
```

Comment:
You might also be able to clean up this test a bit, perhaps by merging the for loop with the example above?  Also, does it make sense to try with each of these scheduler start hosts in particular?  I get the copy-paste history, but we should also do some thinking about cases that might be more relevant to your situation.

Reply:
I've given it another go, merging it with the test from which it was copy-pasting, and also reducing the amount of copy-paste in that test while I'm at it.